File: Python/Land.py
from Plant import Plant

class Land:
    def __init__(self, id_land=0, land_location=""):
        self.id_land = id_land
        self.land_location = land_location
        self.plant_list = []

    # Getter methods
    def get_id_land(self):
        return self.id_land

    def get_land_location(self):
        return self.land_location

    def get_plant_list(self):
        return self.plant_list

    # Setter methods
    def set_land_location(self, land_location):
        self.land_location = land_location

    def set_plant_list(self, daftar):
        self.plant_list = daftar

    # Tambah Plant ke list
    def add_plant(self, plant: Plant):
        self.plant_list.append(plant)

    # Display Land info
    def display_land(self, indent=0):
        sp = " " * indent
        print(f"{sp}Land ID       : {self.get_id_land()}")
        print(f"{sp}Land Location : {self.get_land_location()}\n")

        if self.plant_list:
            print(f"{sp}    Plant:")
            for plant in self.plant_list:
                plant.display_plant(indent + 8)

    def __del__(self):
        pass

File: Java/Land.java
import java.util.ArrayList;
import java.util.List;

public class Land {
    private int idLand;
    private String landLocation;
    private List<Plant> plantList;

    // Default constructor
    public Land() {
        this.idLand = 0;
        this.landLocation = "";
        this.plantList = new ArrayList<>();
    }

    // Parameterized constructor
    public Land(int idLand, String landLocation) {
        this.idLand = idLand;
        this.landLocation = landLocation;
        this.plantList = new ArrayList<>();
    }

    // Getter methods
    public int getIdLand() {
        return idLand;
    }

    public String getLandLocation() {
        return landLocation;
    }

    public List<Plant> getPlantList() {
        return plantList;
    }

    // Setter methods
    public void setLandLocation(String landLocation) {
        this.landLocation = landLocation;
    }

    public void setPlantList(List<Plant> daftar) {
        this.plantList = daftar;
    }

    // Tambah Plant ke list
    public void addPlant(Plant plant) {
        this.plantList.add(plant);
    }

    // Display method
    public void displayLand(int indent) {
        String sp = " ".repeat(indent);
        System.out.println(sp + "Land ID       : " + getIdLand());
        System.out.println(sp + "Land Location : " + getLandLocation());
        System.out.println();

        if (!plantList.isEmpty()) {
            System.out.println(sp + "    Plant:");
            for (Plant plant : plantList) {
                plant.displayPlant(indent + 8);
            }
        }
    }
}
